ballerina/Ballerina.toml

```diff
@@ -9,84 +9,94 @@ icon = "icon.png"
 repository = "https://github.com/ballerina-platform/module-ballerinax-redis"
 license = ["Apache-2.0"]
 
-[[platform.java11.dependency]]
-path = "../native/build/libs/redis-native-2.5.1.jar"
+[platform.java17]
+graalvmCompatible = true
+
+[[platform.java17.dependency]]
 groupId = "org.ballerinalang"
 artifactId = "redis-utils"
 module = "redis-utils" 
 version="2.5.1"
+path = "../native/build/libs/redis-native-2.5.1.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.lettuce"
 artifactId = "lettuce-core"
 version = "5.1.2.RELEASE"
 
 [[platform.java11.dependency]]
 groupId = "org.apache.commons"
 artifactId = "commons-pool2"
-version = "2.4.3"
+version = "2.11.1"
 
-[[platform.java11.dependency]]
-groupId = "com.github.kstyrc"
-artifactId = "embedded-redis"
-version = "0.6"
-
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "com.google.guava"
 artifactId = "guava"
 version = "32.0.0-jre"
+path = "./lib/guava-32.0.0-jre.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "commons-io"
 artifactId = "commons-io"
 version = "2.7"
+path = "./lib/commons-io-2.7.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-common"
 version = "4.1.94.Final"
+path = "./lib/netty-common-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-buffer"
 version = "4.1.94.Final"
+path = "./lib/netty-buffer-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-transport"
 version = "4.1.94.Final"
+path = "./lib/netty-transport-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-resolver"
 version = "4.1.94.Final"
+path = "./lib/netty-resolver-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-handler"
 version = "4.1.94.Final"
+path = "./lib/netty-handler-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-transport-native-kqueue"
 version = "4.1.94.Final"
+path = "./lib/netty-transport-native-kqueue-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-transport-native-epoll"
 version = "4.1.94.Final"
+path = "./lib/netty-transport-native-epoll-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
```

Comment:
What is the conclusion for managing the transitive native dependencies in the repo itself?

Reply:
Had a discussion with @Dilhasha and it seems that declaring the transitive native dependencies in the Ballerina.toml is the recommended options to minimize dependency conflicts.